File: python/talongym/license_notice.py
"""GPL startup notice for TalonGym entry points."""

from __future__ import annotations

import sys

from talongym import __version__
from talongym.credits import CREDIT_LINE

NOTICE = (
    f"TalonGym {__version__} — Copyright (C) 2026 TalonGym contributors.\n"
    f"{CREDIT_LINE}\n"
    "Licensed under GNU GPL v3 or later; see LICENSE in the repository.\n"
    "This program comes with ABSOLUTELY NO WARRANTY."
)

_emitted = False


def emit_license_notice(stream=None) -> None:
    """Print the GPL notice once per process at program startup."""
    global _emitted
    if _emitted:
        return
    _emitted = True
    out = stream or sys.stderr
    print(NOTICE, file=out, flush=True)

File: web/src/licenseNotice.ts
export const CREDIT_LINE =
  "Built by FTC Team 17986 — 904 Robo Eagles and Team 27268 — Talon Strike.";

export const LICENSE_NOTICE =
  "TalonGym — Copyright (C) 2026 TalonGym contributors. " +
  `${CREDIT_LINE} ` +
  "Licensed under GNU GPL v3 or later; see LICENSE in the repository.";

export function logLicenseNotice(): void {
  console.info(LICENSE_NOTICE);
}
